moflask/requests/auth.py

```diff
@@ -0,0 +1,45 @@
+"""Auth middlewares for the requests library."""
+
+import urllib.parse
+
+import flask
+import requests
+
+from . import rest
+
+
+class AuthAppClient(rest.Client):
+    """Client for the impact-stack auth app."""
+
+    @classmethod
+    def from_config(cls):
+        """Create a new auth-app client instance from the app config."""
+        return cls(flask.current_app.config.get("IMPACT_STACK_AUTH_APP_URL"))
+
+    def get_token(self, organization):
+        """Get a JWT token for a sub-organization."""
```

Comment:
I do find "sub-organization" a bit confusing: sub of what, the AUTH app instance? :thinking:

Reply:
Of impact-stack, which is the top-level organization.

Reply:
What do I do if I need a token for the organization the API key is for?

Reply:
> What do I do if I need a token for the organization the API key is for?

Then you have find a new use-case.

Reply:
This sounds like the top-level organization is something different to what we normally call an "organization"? In that case I think the comment should just say "organization" :wink:

Reply:
I think it’s perfectly fine within the domain language we have within impact-stack / auth-app. At the moment we have  impact-stack as the top-level *organization* and all clients as *sub-organizations* and chapters as *sub-sub-organizations* …

Reply:
Ok, I've never heard of it before :shrug: